site-src/implementations.md

```diff
@@ -7,6 +7,8 @@ Implementors and integrators of Gateway API are encouraged to update this
 document with status information about their implementations, the versions they
 cover, and documentation to help users get started.
 
+[View a table to quickly compare supported features of projects](/implementation-table)
```

Comment:
This might benefit from using some kind of callout so it's more noticeable (https://gateway-api.sigs.k8s.io/contributing/style-guide/#callouts)